python/lsst/daf/butler/script/queryDatasets.py

```diff
@@ -46,21 +45,15 @@
 _LOG = logging.getLogger(__name__)
 
 
-@dataclasses.dataclass(frozen=True)
-class _RefInfo:
-    datasetRef: DatasetRef
-    uri: str | None
-
-
 class _Table:
     """Aggregates rows for a single dataset type, and creates an astropy table
     with the aggregated data. Eliminates duplicate rows.
     """
 
-    datasetRefs: set[_RefInfo]
+    datasetRefs: dict[DatasetRef, str | None]
```

Comment:
Is the intent here to drop rows where a ref is associated with more than one URI?  I think that's the effect of this change.  If that's undesirable I think we could safely switch to `list[_RefInfo]` instead of `set[_RefInfo]` now.

Reply:
Good point. I thought it was for handling the deduplication of results (and I now need to retain the returned sorting order) but it also needs to support disassembled composites so I need to check that by querying pipelines_check.

Reply:
It was working already because the dataset type name included the component and there was a table per component. I've rewritten the other code that calls this to generate a table as data arrive rather than accumulating in memory and now `butler query-datasets` starts sending output as soon as the first dataset type results arrives.

Reply:
What about multiple refs from chained datastores?

Reply:
Ok. I have no means of testing this but I do have a fix that I made that stored lists of URIs that I can resurrect (I deleted it when I convinced myself this was only disassembled composites).

Reply:
Ok. I  took a deeper look at this:

* `butler.get_many_uris` only returns one URI per ref so there can't be multiple URIs in this table.
* There is a chaining test that was working by luck and assuming the second datastore would return results. This was luck because there was a bug in the test in that the in-memory datastore was not being used because the butler instance got remade -- fixing that bug causes in-memory dataset URIs to be reported.
* I need to fix get_many_uris so that it won't return ephemeral URIs if permanent ones exist.
* I probably can remove my patch to support multiple URIs since it won't happen.